.ci/lint

```diff
@@ -41,52 +43,117 @@ pipeline {
                 archiveArtifacts artifacts: 'file-list-status.log,file-list-changed.log,file-list-deleted.log,file-list-experiment.log'
             }
         }
-        stage('linting') {
-            failFast false
+        stage('linting chg') {
             parallel {
-                stage('salt linting') {
+                stage('lint salt') {
                     when {
                         expression { return readFile('file-list-changed.log') =~ /(?i)(^|\n)(salt\/.*\.py|setup\.py)\n/ }
                     }
                     steps {
                         sh '''
                         eval "$(pyenv init - --no-rehash)"
-                        grep -Ei '^salt/.*\\.py$|^setup\\.py$' file-list-changed.log | xargs -r '--delimiter=\\n' tox -e pylint-salt | tee pylint-report-salt.log
+                        # tee makes the exit/return code always 0
+                        grep -Ei '^salt/.*\\.py$|^setup\\.py$' file-list-changed.log | (xargs -r '--delimiter=\\n' tox -e pylint-salt;echo"$?" >  pylint-salt-chg.exit) | tee pylint-report-salt.log
                         # remove color escape coding
                         sed -ri 's/\\x1B\\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]//g' pylint-report-salt.log
+                        read rc_exit < pylint-salt-chg.exit
```

Comment:
Im not sure that these are going to work on our build servers.

Why did you choose this command?

Reply:
The above reads the first line of a file.   `read` is just part of the shell commands like `if` i.e.
```
-bash-4.1$ help read
read: read .....
    Read a line from the standard input and split it into fields.
```
Its a lot better method than `rc_exit=$(head -1 pylint-salt-chg.exit)`   # and read removes the line feed

Reply:
I was referring to rc_exit. Not sure that is available in Cent7. Isn't that a Suse specific command?

Reply:
Going to test this out on your PR and see what we get.

Reply:
Or are you just using that as a variable?

Reply:
NM I see it's becoming a variable.